internal/service/directconnect/macsec_key.go

```diff
@@ -0,0 +1,173 @@
+package directconnect
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceMacSecKey() *schema.Resource {
+	return &schema.Resource{
+		// MacSecKey resource only supports create (Associate), read (Describe) and delete (Disassociate)
+		Create: resourceMacSecKeyCreate,
+		Read:   resourceMacSecKeyRead,
+		// You cannot modify a MACsec secret key after you associate it with a connection.
+		// To modify the key, disassociate the key from the connection, and then associate
+		// a new key with the connection
+		Update: schema.Noop,
+		Delete: resourceMacSecKeyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cak": {
+				Type:     schema.TypeString,
+				Optional: true,
+				// CAK requires CKN
+				RequiredWith: []string{"ckn"},
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`[a-fA-F0-9]{64}$`), "Must be 64-character hex code string"),
+			},
+			"ckn": {
+				Type:         schema.TypeString,
+				Computed:     true,
```

Comment:
When creating a MACsec key with the `SecretArn` parameter, `ckn` will be returned, so I believe this field needs to be computed. Otherwise, when running terraform plan, Terraform will return a diff between the blank field in the HCL and the `ckn` value in the state file, and prompt for an update.